core/src/KokkosExp_MDRangePolicy.hpp

```diff
@@ -162,7 +179,7 @@ struct MDRangePolicy : public Kokkos::Impl::PolicyTraits<Properties...> {
   enum { rank = static_cast<int>(iteration_pattern::rank) };
 
   using index_type       = typename traits::index_type;
-  using array_index_type = long;
+  using array_index_type = std::int64_t;
```

Comment:
This change was suggested by @crtrott 
It does not absolutely have to be part of this PR.

Reply:
Cabana was building `MDRangePolicy` with arrays of long without setting the index type on the policy.
Although this was invalid use we must recognize this change may break code downstream.